com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/parser/DockerInfoParser.java

```diff
@@ -0,0 +1,49 @@
+package com.sap.cloud.lm.sl.cf.core.parser;
+
+import java.util.List;
+import java.util.Map;
+
+import org.cloudfoundry.client.lib.domain.DockerCredentials;
+import org.cloudfoundry.client.lib.domain.DockerInfo;
+
+public class DockerInfoParser implements ParametersParser<DockerInfo> {
+   
+    private static final String DOCKER = "docker";
+    
+    @Override
+    public DockerInfo parse(List<Map<String, Object>> parametersList) {
+        Map<String, String> dockerParams = getDockerParams(parametersList);
+        DockerInfo dockerInfo = getDockerInfo(dockerParams);
+        
+        return dockerInfo;
+    }
+    
+    @SuppressWarnings("unchecked")
+    private Map<String, String> getDockerParams(List<Map<String, Object>> parametersList) {
+       return (Map<String, String>) parametersList.stream()
+            .filter(p -> p.get(DOCKER) != null)
+            .findFirst()
+            .map(p -> p.get(DOCKER))
+            .orElse(null);
+    }
+    
+    private DockerInfo getDockerInfo(Map<String, String> docker) {
+        String image = null;
```

Comment:
Apply the [return-early pattern](http://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html) in this function.